### PR TITLE
TST: enable pyogrio datetime tz tests

### PR DESCRIPTION
**Pull request by m-richards**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Some unit (ms vs ns) differences that need to be taken into account, it seems

File: geopandas/io/file.py [hunk 1]
```diff
@@ -405,7 +406,10 @@ def _read_file_fiona(
                 # fiona only supports up to ms precision (any microseconds are
                 # floating point rounding error)
                 if not (as_dt.dtype == "object"):
-                    df[k] = as_dt.dt.round(freq="ms")
+                    if PANDAS_GE_20:
+                        df[k] = as_dt.dt.as_unit("ms")
+                    else:
+                        df[k] = as_dt.dt.round(freq="ms")
```
Comment (m-richards) [inline geopandas/io/file.py:412]: Perhaps I should do this as a separate PR. Basically pyogrio now uses the ms resolution pandas 2.0 datetime dtype, but fiona does not. Seemed better in terms of outcome to make fiona IO do this as well, instead of having conditional testing.

Comment (jorisvandenbossche) [reply geopandas/io/file.py:412]: I am fine with following the behaviour of pyogrio here. Although we still won't support out of bounds (which we then in theory could), because we first still use `pd.to_datetime` to parse the strings to nanosecond-resolution timestamps ..

**Review by jorisvandenbossche — APPROVED** [review]
I think this is good to go?

File: geopandas/io/file.py (hunk 1 above)
Comment (m-richards) [reply geopandas/io/file.py:412]: Let's leave that as a follow up since that would actually change behaviour in those cases.

I think this is ready, unless we think the ms units for pandas 2.0 fiona warrants a changelog entry (I don't really think it does)

**Comment by jorisvandenbossche** [comment]
Thanks!
